### PR TITLE
LOK-1887: handle azure monitoring edge cases

### PR DESCRIPTION
**Pull request by freddy61025**
## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-1887

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Review by vserrago — APPROVED** [review]
Overall LGTM, but with some open questions

File: minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureScanner.java [hunk 1]
```diff
@@ -211,12 +211,38 @@ private AzureScanItem scanNetworkInterfaces(AzureScanItem scanItem,
...
+        }
+    }
+
     private List<AzureNetworkInterface> findNetworkInterfacesForVmId(AzureNetworkInterfaces networkInterfaces, String vmId) {
         List<AzureNetworkInterface> networkInterfacesList = new ArrayList<>();
         for (AzureNetworkInterface networkInterface : networkInterfaces.getValue()) {
             NetworkInterfaceProps properties = networkInterface.getProperties();
+            if (properties == null) {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (freddy61025) [reply minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureScanner.java]: added

File: minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureScanner.java [hunk 2]
```diff
@@ -169,13 +176,19 @@ private AzureScanItem scanNetworkInterfaces(AzureScanItem scanItem,
         List<AzureScanNetworkInterfaceItem> scannedNetworkInterfaces = new ArrayList<>();
 
         for (AzureNetworkInterface networkInterface : interfaceList) {
-
             NetworkInterfaceProps networkInterfaceProps = networkInterface.getProperties();
+            if (networkInterfaceProps == null) {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (freddy61025) [reply minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureScanner.java]: added

File: minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureCollector.java [hunk 3]
```diff
@@ -115,20 +117,15 @@ public CompletableFuture<CollectionSet> collect(CollectionRequest collectionRequ
...
                     .collect(Collectors.toCollection(ArrayList::new));
 
                 // interface metrics
-                for (var resources : request.getCollectorResourcesList()) {
-                    metricResults.addAll(collectInterfaceMetrics(request, resources, token)
-                        .entrySet().stream().map(interfaceMetric ->
-                            mapInterfaceResult(request, resources.getResource(), resources.getType(), interfaceMetric))
-                        .toList());
-                }
+                this.collectNetworkMetrics(request, token, metricResults);
```
Comment (cgorantla) [inline minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureCollector.java:128]: nit: may not need `this` here.

File: minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureScanner.java [hunk 4]
```diff
@@ -93,7 +94,9 @@ public CompletableFuture<ScanResultsResponse> scan(Any config) {
 
             for (var resourceGroupValue : resourceGroups.getValue()) {
                 String resourceGroup = resourceGroupValue.getName();
-                scannedItems.addAll(scanForResourceGroup(request, token, resourceGroup));
+                if (StringUtils.isNotEmpty(resourceGroup)) {
```
Comment (cgorantla) [inline minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureScanner.java:97]: Our preference for these kinds of utils is Guava.  
Did you check apache lang3 was already part of karaf features and this bundle loads properly ?

**Review by cgorantla — APPROVED** [review]

File: minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureScanner.java (hunk 4 above)
Comment (freddy61025) [reply minion/plugins/azure-plugin/src/main/java/org/opennms/horizon/minion/azure/AzureScanner.java]: changed

**Pull request merged**